include/kernel.h

```diff
@@ -1318,78 +1318,6 @@ __syscall const char *k_thread_name_get(k_tid_t thread_id);
  * @}
  */
 
-/**
- * @addtogroup clock_apis
- * @{
- */
-
-/**
- * @brief Generate null timeout delay.
- *
- * This macro generates a timeout delay that that instructs a kernel API
- * not to wait if the requested operation cannot be performed immediately.
- *
- * @return Timeout delay value.
- */
-#define K_NO_WAIT 0
```

Comment:
why is it moved? i think that it would be better to keep it here (part of kernel API). Internals (complexity) could be hidden in `sys_clock.h` and here have something like `#define K_NO_WAIT Z_NO_WAIT` same with `K_FOREVER`. I would also prefer to see conversion macros here as well. They can also be simplified to `#define K_TIMEOUT_MS(ms) Z_TIMEOUT_MS(ms)`.